app/utilities/filterCompanyEmployees.js

```diff
@@ -0,0 +1,16 @@
+export function filterCompanyEmployees(fetchedCompanies, fetchedUsers) {
+  function filterUsersByCompany(users, companyUri) {
+    return users.filter(function (user) {
+      return user.uris.company === companyUri;
+    });
+  }
+
+  const combinedArrays = fetchedCompanies.map(function (company) {
+    return {
+      ...company,
+      employees: filterUsersByCompany(fetchedUsers, company.uri),
```

Comment:
You're nesting two loops here. You're doing `filter` inside `map`. This can be done without nesting two loops inside each other.

If you want to get the data of a company with a particular id when you have an array of companies, you need a loop. 
However, it changes if you have a dictionary.

First, change the companies into a dictionary where the id is the key.
(they don't have to be sorted yet)

```
{
  0: {
    name: "Company 0",
    uri: "/companies/0",
  },
  1: {
    name: "Company 1",
    uri: "/companies/1",
  },
}
```
Now, getting the data of a company with id `10` is as simple ass doing `companiesDictionary[10]`.

Then, add the `employees` property.
```
{
  0: {
    name: "Company 0",
    uri: "/companies/0",
    employees: [
      {
        name: "User 601",
        uri: "/users/601",
        email: "user601@company0.com",
        uris: { "company": "/companies/0" }
      },
      // ...
    ]
  },
  1: {
    name: "Company 1",
    uri: "/companies/1",
    "employees": [
       {
        name: "User 343",
        uri: "/users/343",
        email: "user343@company1.com",
        uris: { "company": "/companies/1" }
      },
      // ...
    ]
  },
  // ...
}
```

Then you can easily change it back to an array of companies using `Object.values`